linalg/solvers.cpp

```diff
@@ -1319,6 +1319,54 @@ void NewtonSolver::Mult(const Vector &b, Vector &x) const
    final_norm = norm;
 }
 
+void NewtonSolver::CheckJacobian(Vector &x, Array<int> &ess_tdof)
+{
+   MFEM_ASSERT(oper != nullptr, "Call SetOperator first.");
+
+   double eps = 1e-8;
+   Vector xpert(height), one(height), fx(height),
+          fxp(height), basis(height), jac_col(height);
+
+   basis = 0.0;
+   xpert = x;
+
+   Operator &jac = oper->GetGradient(x);
+
+   oper->Mult(x, fx);
+
+   for (int j = 0; j < height; j++)
+   {
+      // Is column associated with an essential boundary?
+      if (ess_tdof.FindSorted(j) >= 0)
```

Comment:
this method is expensive, there's no need to use this optimization and assume ess_tdof is sorted